onnxruntime/core/graph/graph_utils.cc

```diff
@@ -715,6 +715,17 @@ void FinalizeNodeFusion(Graph& graph, const std::vector<std::reference_wrapper<N
   }
 }
 
+void FinalizeNodeFusion(Graph& graph, const std::vector<std::reference_wrapper<Node>>& nodes, Node& replacement_node_start,
+Node& replacement_node_end) {
+  MoveAllNodeInputEdges(graph, nodes.front(), replacement_node_start);
+  MoveAllNodeOutputs(graph, nodes.back(), replacement_node_end);
+
+  for (Node& node : nodes) {
+    RemoveNodeOutputEdges(graph, node);
```

Comment:
>RemoveNodeOutputEdges [](start = 4, length = 21)

Would this removal cause missing input to other nodes outside `nodes`?

Reply:
If any of the nodes in `nodes` have consumers that are not being removed, the fusions should not be triggered. The fusion implementations should already check for this case